torch/export/_unlift.py

```diff
@@ -309,6 +318,19 @@ def _create_stateful_graph_module(
     for const_name, value in ep.constants.items():
         if not torch.fx.graph_module._has_attr(stateful_gm, const_name):
             if isinstance(value, torch.Tensor):
+                if value.requires_grad:
+                    warnings.warn(
+                        f"A model attribute `{const_name}` requires gradient. "
+                        f"but it's not properly registered as a parameter. "
+                        f"torch.export will detach it and treat it as a constant tensor "
+                        f"but please register it as parameter instead."
+                    )
+                    if value in original_tensor_to_detached_tensor:
+                        value = original_tensor_to_detached_tensor[value]
+                    else:
+                        detached_value = value.detach()
```

Comment:
maybe this code should be refactored into a util?

Reply:
It is bit hard to make it a util because this is only relevant in this specific location.